pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -171,7 +173,8 @@ public SuccessResponse addSchema(
   public String validateSchema(FormDataMultiPart multiPart) {
     Schema schema = getSchemaFromMultiPart(multiPart);
     try {
-      SchemaUtils.validate(schema);
+      List<TableConfig> tableConfigs = getTableConfigsForSchema(schema.getSchemaName());
```

Comment:
I wonder if this (along with the getTableConfigsForSchema function) can be moved inside validate ? So that the caller doesn't have to do this every single time (and can be done implicitly)

Reply:
i thought about that. I was just trying to avoid passing pinotHelixResourceManager to the utils methods. I can change it to that if you think that's better. I was going back and forth about it

Reply:
Actually, as I tried to do this, I realized that PinotHelixResourceManager is in pinot-controller, and the utils (which are in pinot-core) do not have access to it. We'd have to pass propertyStore, or change the design of SchemaUtils so we can inject and keep state.
Now I think it's better that the caller fetches what it needs. Practically,  PinotTableRestletResource and PinotSchemaRestletResource should be the only places adding/modifying table/schema.

Reply:
Understood. Thanks for looking into this.